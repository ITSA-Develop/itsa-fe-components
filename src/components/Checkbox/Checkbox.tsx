import { Checkbox as AntCheckbox, CheckboxProps } from 'antd';

export interface ICheckboxProps extends CheckboxProps {
	variant?: 'default' | undefined;
}

export const Checkbox = ({ ...rest }: ICheckboxProps) => {
	const { variant, className, rootClassName, ...props } = rest;
	const mergedRootClassName = [
		rootClassName,
		variant === 'default' ? 'itsa-checkbox--default' : undefined,
	]
		.filter(Boolean)
		.join(' ');

	return (
		<AntCheckbox
			{...props}
			className={className}
			rootClassName={mergedRootClassName}
		/>
	);
};
