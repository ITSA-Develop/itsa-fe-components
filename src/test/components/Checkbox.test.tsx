import { fireEvent, render, screen } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';
import { Checkbox } from '../../components/Checkbox/Checkbox';

describe('Checkbox component', () => {
	it('renders with label text', () => {
		const { container } = render(<Checkbox>Check me</Checkbox>);
		expect(screen.getByText('Check me')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('calls onChange when clicked', () => {
		const onChange = vi.fn();
		const { container } = render(<Checkbox onChange={onChange}>Click me</Checkbox>);
		const checkbox = screen.getByRole('checkbox');
		fireEvent.click(checkbox);
		expect(onChange).toHaveBeenCalled();
		expect(container).toMatchSnapshot();
	});

	it('is checked when defaultChecked is true', () => {
		const { container } = render(<Checkbox defaultChecked>Checked</Checkbox>);
		const checkbox = screen.getByRole('checkbox') as HTMLInputElement;
		expect(checkbox.checked).toBe(true);
		expect(container).toMatchSnapshot();
	});

	it('can be disabled', () => {
		const { container } = render(<Checkbox disabled>Disabled</Checkbox>);
		const checkbox = screen.getByRole('checkbox');
		expect(checkbox).toBeDisabled();
		expect(container).toMatchSnapshot();
	});
});
