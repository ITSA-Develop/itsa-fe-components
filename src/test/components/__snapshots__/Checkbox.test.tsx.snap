// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Checkbox component > calls onChange when clicked 1`] = `
<div>
  <label
    class="ant-checkbox-wrapper css-dev-only-do-not-override-l9pxc0"
  >
    <span
      class="ant-checkbox ant-wave-target css-dev-only-do-not-override-l9pxc0 ant-checkbox-checked"
    >
      <input
        class="ant-checkbox-input"
        type="checkbox"
      />
      <span
        class="ant-checkbox-inner"
      />
    </span>
    <span
      class="ant-checkbox-label"
    >
      Click me
    </span>
  </label>
</div>
`;

exports[`Checkbox component > can be disabled 1`] = `
<div>
  <label
    class="ant-checkbox-wrapper ant-checkbox-wrapper-disabled css-dev-only-do-not-override-l9pxc0"
  >
    <span
      class="ant-checkbox ant-wave-target css-dev-only-do-not-override-l9pxc0 ant-checkbox-disabled"
    >
      <input
        class="ant-checkbox-input"
        disabled=""
        type="checkbox"
      />
      <span
        class="ant-checkbox-inner"
      />
    </span>
    <span
      class="ant-checkbox-label"
    >
      Disabled
    </span>
  </label>
</div>
`;

exports[`Checkbox component > is checked when defaultChecked is true 1`] = `
<div>
  <label
    class="ant-checkbox-wrapper css-dev-only-do-not-override-l9pxc0"
  >
    <span
      class="ant-checkbox ant-wave-target css-dev-only-do-not-override-l9pxc0 ant-checkbox-checked"
    >
      <input
        checked=""
        class="ant-checkbox-input"
        type="checkbox"
      />
      <span
        class="ant-checkbox-inner"
      />
    </span>
    <span
      class="ant-checkbox-label"
    >
      Checked
    </span>
  </label>
</div>
`;

exports[`Checkbox component > renders with label text 1`] = `
<div>
  <label
    class="ant-checkbox-wrapper css-dev-only-do-not-override-l9pxc0"
  >
    <span
      class="ant-checkbox ant-wave-target css-dev-only-do-not-override-l9pxc0"
    >
      <input
        class="ant-checkbox-input"
        type="checkbox"
      />
      <span
        class="ant-checkbox-inner"
      />
    </span>
    <span
      class="ant-checkbox-label"
    >
      Check me
    </span>
  </label>
</div>
`;
